### PR TITLE
Hitboxes

### PR DESCRIPTION
**Pull request by raphael-pietrzak**
## Summary

This PR implements enhancements and fixes related to hitboxes, NPC movement, and entities in the game.

- Enhanced the hitbox functionality for better collision detection.
- Main layer based on bottom hitbox.
- Created a test function to draw rects and hitboxes when the [H] key is pressed.
- Draw apples with the tree.

## Media

350527460-b6db8cd7-35e0-4426-89d2-5876334b9dc3
350527474-470f6ee4-f68a-4cb2-9020-e0c85eaf4089

## Checklist

- [x] I have tested this change locally and it works as expected.
- [x] I have made sure that the code follows the formatting and style guidelines of the project.

## Labels

`type: enhancement`, `area: hitboxes`, `game-npc`, `game-entities`

### CODE REVIEW

File: src/groups.py
Comment (fortwoone) [inline src/groups.py:30]: This change makes it so text boxes do not get displayed on the screen regardless of the player’s position anymore.

File: =
Comment (fortwoone) [inline =:1]: Uh, what is this file?

**Review by fortwoone — CHANGES_REQUESTED** [review]
In general, this looks good. But some nitpicks prevent it from being merged. Also, a rebase is required.

File: src/groups.py
Comment (raphael-pietrzak) [reply src/groups.py:30]: i need to create draw function in textBoxes good catch

File: =
Comment (raphael-pietrzak) [reply =:1]: i don't know lol

File: src/groups.py
Comment (fortwoone) [reply src/groups.py:30]: …or you could just revert this change

Comment (raphael-pietrzak) [reply src/groups.py:29]: he thing is, the text boxes are not part of the AllSprites group, so I don’t understand why those lines were there. It’s never going to display the text boxes anyway since they are not part of it; they are part of a list _tb_list.

Comment (fortwoone) [reply src/groups.py:29]: Actually, they are. Which is why these lines are there

File: src/npc/dialog.py
Comment (fortwoone) [inline src/npc/dialog.py:197]: No, you added that to the DialogueManager which is not a sprite

**Review by fortwoone — CHANGES_REQUESTED** [review]

File: src/groups.py
Comment (raphael-pietrzak) [reply src/groups.py:29]: Oh yes, sorry, I just saw that.

So my question is, why doesn’t the TextBox class inherit from the Sprite class (not the pygame one)?

Normally, it should behave the same as with the isinstance check with the draw function I added in TextBox. How do I check if they work because, even before, I never saw them?

Comment (fortwoone) [reply src/groups.py:29]: The non-pygame Sprite base class requires parameters that are useless in the context of displaying textboxes

Comment (raphael-pietrzak) [reply src/groups.py:29]: okok

File: src/npc/dialog.py
Comment (fortwoone) [inline src/npc/dialog.py:43]: I put that value in a constant because it wasn’t needed elsewhere, which is why I will ask you to revert that change

**Review by fortwoone — CHANGES_REQUESTED** [review]
There is something to revert here, and a rebase is required before I can give my approval, due to merge conflicts

File: src/npc/dialog.py
Comment (raphael-pietrzak) [reply src/npc/dialog.py:43]: I think we just have different ideas of what clear code should be, but okay, I’ll revert

**Review by fortwoone — APPROVED** [review]
LGTM. You just need to fetch-rebase onto main now

**Pull request closed without merging**
